support/cas-server-support-validation-core/src/main/java/org/apereo/cas/web/AbstractServiceValidateController.java

```diff
@@ -218,11 +218,15 @@ protected ModelAndView handleTicketValidation(final HttpServletRequest request,
         var proxyIou = StringUtils.EMPTY;
         val proxyHandler = serviceValidateConfigurationContext.getProxyHandler();
         if (serviceCredential != null && proxyHandler != null && proxyHandler.canHandle(serviceCredential)) {
-            proxyIou = handleProxyIouDelivery(serviceCredential, proxyGrantingTicketId);
-            if (StringUtils.isEmpty(proxyIou)) {
-                val description = getTicketValidationErrorDescription(CasProtocolConstants.ERROR_CODE_INVALID_PROXY_CALLBACK,
-                    new Object[]{serviceCredential.getId()}, request);
-                return generateErrorView(CasProtocolConstants.ERROR_CODE_INVALID_PROXY_CALLBACK, description, request, service);
+            val registeredService = ((HttpBasedServiceCredential) serviceCredential).getService();
+            val authorizedToReleaseProxyGrantingTicket = registeredService.getAttributeReleasePolicy().isAuthorizedToReleaseProxyGrantingTicket();
+            if (!authorizedToReleaseProxyGrantingTicket) {
+                proxyIou = handleProxyIouDelivery(serviceCredential, proxyGrantingTicketId);
+                if (StringUtils.isEmpty(proxyIou)) {
+                    val description = getTicketValidationErrorDescription(CasProtocolConstants.ERROR_CODE_INVALID_PROXY_CALLBACK,
+                            new Object[]{serviceCredential.getId()}, request);
+                    return generateErrorView(CasProtocolConstants.ERROR_CODE_INVALID_PROXY_CALLBACK, description, request, service);
+                }
```

Comment:
A log message here explaining the behavior at DEBUG level would be very helpful.

Reply:
Sure. This will help.